### PR TITLE
vim-patch:8.1.0228, 8.1.1384, 8.1.1386, 8.1.1393, 8.2.3040

### PR DESCRIPTION
**Pull request by VVKot**
Drag&drop patches are N/A, there are no relevant changes. Ported several small changes from the memory allocation/type casts patches - those places and everything else already uses `xmalloc`. Comments added in those patches are also N/A because Neovim uses standard lib like `memcpy` instead of custom memory allocation.

### CODE REVIEW

File: src/nvim/fileio.c [hunk 1]
```diff
@@ -101,7 +101,7 @@ struct bw_info {
   int bw_restlen;                // nr of bytes in bw_rest[]
   int bw_first;                  // first write call
   char_u *bw_conv_buf;      // buffer for writing converted chars
-  int bw_conv_buflen;            // size of bw_conv_buf
+  size_t bw_conv_buflen;            // size of bw_conv_buf
```
Comment (bfredl) [inline src/nvim/fileio.c:104]: adjust spaces for readability (also the one above, as someone forgot it then)

Comment (bfredl) [inline src/nvim/fileio.c:104]: adjust spaces for readability (also the one above, as someone forgot it then)

**Comment by janlazo** [comment]
vim-patch.sh assumes non-empty `vim-patch:` commits. Move the empty N/A commits inside the commit description of a non-empty vim-patch commit.

**Comment by VVKot** [comment]
@janlazo thanks for the tip, should be good now!

**Pull request merged**